### ISSUE TITLE
Faster Dev Server Boot

### ISSUE THREAD

**Issue by mcrascal** [evidence-dev/evidence#1564]
npm run dev takes _a lot_ longer to get to a live app than a bare sveltekit app does.

**Comment by csjh** [comment]
Some profiling suggests a major cause is large files in dependencies (~4.1s). Main perpetrators:
<img width="1368" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMk9EQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a6ab29c0ecdbd0ac1aeb14d41731671611bb47d2/image.png" alt="image.png" />
<img width="1352" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMlNEQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--e526bf116bc45ff3adc3468cc99739d64a7e7eff/image.png" alt="image.png" />
<img width="1380" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMmVEQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--118f2ef97dee2ace1e61444d61d005a2902ba6d1/image.png" alt="image.png" />
There's a few other fat time sinks like svelte file compilation (~1.7s) - 170 files are compiled when the dev server is started, so we should figure out if there's a way to avoid that.

**Comment by archiewood** [comment]
Can't actually see these images, but I assume they are inadvertent additions?

**Comment by csjh** [comment]
Whoops, updated

**Comment by hughess** [comment]
I'm looking through vite config for something else and stumbled upon this warmup option - sharing in case it's useful here:
https://vitejs.dev/config/server-options.html#server-warmup

**Comment by csjh** [comment]
Yeah seems like that helps move the heavy bits to the start of the dev server instead of later in, but the end result is the same

**Comment by archiewood** [comment]
@csjh what was your rough profiling setup to get these measurements?

**Comment by csjh** [comment]
@archiewood I've been modifying `node_modules/vite/bin/vite.js` because their `--profile` flag wasn't working 100% correctly for me

Basically just go in that file and replace
```js
  session.post('Profiler.enable', () => {
    session.post('Profiler.start', start)
  })
```
(located near the bottom) with
```js
  session.post('Profiler.enable', () => {
    session.post('Profiler.start', start)
	setTimeout(() => {
		session.post('Profiler.stop', async (err, { profile }) => {
			if (err) {
				console.error(err)
				process.exit(1)
			}
			const fs = await import('node:fs')
			const path = await import('node:path')
			const filename = path.resolve(process.cwd(), 'v8-profile.cpuprofile')
			fs.writeFileSync(filename, JSON.stringify(profile))
			console.log(`Profile written to ${filename}`)
			process.exit(0)
		})
	}, 20000);
  })
```
and run vite with the `--profile` flag
after the `.cpuprofile` can be loaded into chrome
